exporter/opentelemetry-exporter-prometheus/src/opentelemetry/exporter/prometheus/__init__.py

```diff
@@ -228,17 +230,17 @@ def _translate_to_prometheus(
 
             pre_metric_family_ids = []
 
-            metric_name = ""
-            metric_name += self._sanitize(metric.name)
+            metric_name = sanitize_full_name(metric.name)
 
             metric_description = metric.description or ""
+            metric_unit = map_unit(metric.unit)
```

Comment:
Should we add a flag for this to allow users to keep the opt-out and use the old behavior?

Reply:
Like, an environment variable?

Reply:
Would this be as a large of a change as let's say...http semantic convention from old to new? Given that this is similarly a beta component (like the instrumentations), I am open to simply making this change and treating it as a specification change (and breaking customers) but if we feel the amount of usage also warrants an opt-in mechanism I am also open to that (although leaning towards just making the change).

Reply:
Wee https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk_exporters/prometheus.md for the configuration recommended by the exporter spec.  There can be an option to disable unit suffixes, but it should be enabled by default if it exists.

Reply:
@dashpole the existing behavior is to always add units without normalizing to prometheus conventions. So the option we're discussing here would just be something to skip the unit mapping and keep the existing behavior.

I think the `without_type_suffix` would be a separate feature.

Reply:
yes, disabling type and unit suffixes will be separate config knobs.

Reply:
You are more than welcome to use env vars to "feature gate" this change to make it easier for users to transition.  That would be orthogonal to the without_units config described in the spec.

Reply:
Added a new **opt-out** environment variable `OTEL_PYTHON_EXPERIMENTAL_DISABLE_PROMETHEUS_UNIT_NORMALIZATION` in 50a3abdc0a6c4ba9cfd3cba48001b070b8f875c0